File: pset6/sentimental-credit/credit.py
def main():
    number = int(input("Number: "))
    temp = number
    sum = 0
    count = 0
    digit = 0

    temp = number
    while temp > 0:
        digit = temp % 10

        if count % 2 == 1:
            doubled = digit * 2
            sum += doubled // 10 + doubled % 10
        else:
            sum += digit

        temp //= 10
        count += 1

    if sum % 10 != 0:
        print("INVALID")
        return

    temp = number
    while temp >= 100:
        temp //= 10
    first_two = temp
    first = first_two // 10

    if (first_two == 34 or first_two == 37) and count == 15:
        print("AMEX")
    elif first_two >= 51 and first_two <= 55 and count == 16:
        print("MASTERCARD")
    elif first == 4 and (count == 13 or count == 16):
        print("VISA")
    else:
        print("INVALID")


main()

File: pset1/credit/credit.c
#include <cs50.h>
#include <stdio.h>


int main(void)
{

    long number = get_long("Number: ");
    long temp = number;

    int sum = 0;
    int count = 0;
    int digit = 0;


    while (temp > 0)
    {
        digit = temp % 10;

        if (count % 2 == 1)
        {
            int doubled = digit * 2;
            sum += doubled / 10 + doubled % 10;
        }
        else
        {
            sum += digit;
        }

        temp /= 10;
        count++;
    }

    if (sum % 10 != 0)
    {
        printf("INVALID\n");
        return 0;
    }

    temp = number;
    while (temp >= 100)
    {
        temp /= 10;
    }
    int first_two = temp;
    int first = first_two / 10;

    if ((first_two == 34 || first_two == 37) && count == 15)
    {
        printf("AMEX\n");
    }
    else if (first_two >= 51 && first_two <= 55 && count == 16)
    {
        printf("MASTERCARD\n");
    }
    else if (first == 4 && (count == 13 || count == 16))
    {
        printf("VISA\n");
    }
    else
    {
        printf("INVALID\n");
    }
}
